### PR TITLE
Use data representative to prevent ..on <Entity> for generic purpose (defaultValues - defaultSecondValues)

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Goals

Data representative for an entity generated by backend and simply used by frontend to prevent ..on <Entity> graphQL listing hell for each entity (e.g. EntityStixCoreRelationshipsEntitiesLine_node fragment in EntityStixCoreRelationshipsEntitiesLine.tsx)

### Steps
- Add stix representative in resolver for StixCoreObject, StixCoreRelationship & StixSightingRelationship
- Use representative in ExternalReferenceStixCoreObjects.jsx
- Write representative as denormalization for From & To (stix-core-relationship & stix-sighting)
- - Use as default mechanism, the name's or type's from and to

### CODE REVIEW

**Comment by RomuDeuxfois** [comment]
Can we write the representative as denormalization ?
Tests :
- Denormalization of name at creation/update/delete ? (if the name of entity change, the denormalization of this entity in relationship is updated ?)

File: opencti-platform/opencti-graphql/src/database/engine.js [hunk 1]
```diff
@@ -2338,7 +2341,7 @@ export const elUpdateElement = async (instance) => {
   // If entity with a name, must update connections
   let connectionPromise = Promise.resolve();
   if (esData.name && isStixObject(instance.entity_type)) {
-    connectionPromise = elUpdateConnectionsOfElement(instance.internal_id, { name: esData.name });
+    connectionPromise = elUpdateConnectionsOfElement(instance.internal_id, { name: esData.name, representative: esData.representative });
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/engine.js]: elUpdateConnectionsOfElement(instance.internal_id, { name: esData.representative });
Keep name for retro-compat

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  return String(secondValue);
+};
+
+// -- RELATIONSHIP --
+
+const extractRelationshipRepresentativeName = (relationshipData) => {
+  let fromRepresentative = relationshipData.fromRepresentative?.main;
+  if (isEmptyField(fromRepresentative)) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]: Always get name property (not fromRepresentative)

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    fromRepresentative = relationshipData.from?.name;
+  }
+  if (isEmptyField(fromRepresentative)) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]: Remove type

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+import moment from 'moment';
+import { isEmptyField, isNotEmptyField } from './utils';
+import { isStixRelationship } from '../schema/stixRelationship';
+import { ENTITY_TYPE_CAPABILITY } from '../schema/internalObject';
+import { isStixCyberObservable } from '../schema/stixCyberObservable';
+import { observableValue } from '../utils/format';
+
+const extractRepresentativeDescription = (entityData) => {
+  let secondValue;
+
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js:10]: if module then retrieveFromModule

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  if (isNotEmptyField(entityData.description)) {
+    secondValue = entityData.description;
+    // BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.x_opencti_description)) {
+    secondValue = entityData.x_opencti_description;
+    // BasicStoreEntity | BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.content)) {
+    secondValue = entityData.content;
+    // BasicStoreBase
+  } else if (isNotEmptyField(entityData.entity_type)) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]: Remove this

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  return String(secondValue);
+};
+
+// -- RELATIONSHIP --
+
+const extractRelationshipRepresentativeName = (relationshipData) => {
+  const mainValue = `${relationshipData.fromName} ➡️ ${relationshipData.toName}`;
+  return String(mainValue);
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js:33]: Remove casting

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    // BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.x_opencti_description)) {
+    secondValue = entityData.x_opencti_description;
+    // BasicStoreEntity | BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.content)) {
+    secondValue = entityData.content;
+  }
+
+  if (isEmptyField(secondValue) || secondValue === 'Unknown') {
+    return '';
```
Comment (Archidoit) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]: isn't it better to return 'null' for an empty description?

**Comment by Archidoit** [comment]
Would be good to replace some "...on XXX { name} " to test the new representative attribute :)

**Comment by Archidoit** [comment]
Relationship new code working well 👍

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  }
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    secondValue = entityData.description;
+    // BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.x_opencti_description)) {
+    secondValue = entityData.x_opencti_description;
+    // BasicStoreEntity | BasicStoreCyberObservable
+  } else if (isNotEmptyField(entityData.content)) {
+    secondValue = entityData.content;
+  }
+
+
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js]:
```suggestion

```

**Comment by RomuDeuxfois** [comment]
> Would be good to replace some "...on XXX { name} " to test the new representative attribute :)

You mean : to check the differences between name and representative ?

@RomuDeuxfois to test the behavior in different places in the platform (ex: ref relationships in Graph, names in notifications or filters etc.

File: opencti-platform/opencti-graphql/src/database/entity-representative.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    secondValue = entityData.content;
+  }
+
+  return secondValue;
+};
+
+// -- RELATIONSHIP --
+
+const extractRelationshipRepresentativeName = (relationshipData) => {
+  return `${relationshipData.fromName} ➡️ ${relationshipData.toName}`;
```
Comment (SarahBocognano) [inline opencti-platform/opencti-graphql/src/database/entity-representative.js:28]: Just one question: What is that arrow ?? How did you do that ?

**Review by SarahBocognano — COMMENTED** [review]
Tested ✅ Everything works perfectly, can't wait to add that through the entire app !

File: opencti-platform/opencti-graphql/src/database/entity-representative.js (hunk 10 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/src/database/entity-representative.js:28]: Emojipedia :D

**Review by richard-julien — APPROVED** [review]

**Pull request merged**